addons/mrp/models/mrp_bom.py

```diff
@@ -180,6 +181,39 @@ def _bom_find_domain(self, product_tmpl=None, product=None, picking_type=None, c
         # order to prioritize bom with product_id over the one without
         return domain
 
+    @api.model
+    def _bom_find_multi(self, products, picking_type=None, company_id=False, bom_type=False):
+        """ Return {product: bom}, product is a product.product record
+        or products are passed in args """
+        if not products:
+            return {}
+        domain = []
+        if picking_type:
+            domain += ['|', ('picking_type_id', '=', picking_type.id), ('picking_type_id', '=', False)]
+        if company_id or self.env.context.get('company_id'):
+            domain += ['|', ('company_id', '=', False), ('company_id', '=', company_id or self.env.context.get('company_id'))]
+        if bom_type:
+            domain += [('type', '=', bom_type)]
+        # Can fetch more than expected (when bom with product_tmpl_id and product_id but only match the template condition), filtered out after
+        boms = self.search(domain + ['|', ('product_id', 'in', products.ids), ('product_tmpl_id', 'in', products.product_tmpl_id.ids)], order='sequence, product_id, id')
```

Comment:
Why do you use domain:
<pre>
'|',
('product_id', 'in', products.ids),
('product_tmpl_id', 'in', products.product_tmpl_id.ids),
</pre>
instead of:
<pre>
'|',
('product_id', 'in', products.ids),
'&', ('product_id', '=', False), ('product_tmpl_id', 'in', products.product_tmpl_id.ids),
</pre>

Reply:
Yes, good point, it simplifies a little bit the code after too. Thx.